src/main/Port.Adapter/UI/Views/Blazor.Common/TreeView.razor

```diff
@@ -316,6 +325,152 @@
+                        if(childNode.ExpansionState == ExpansionState.Expanded && childNode.Children != null && childNode.Children.Any())
+                        {
+                            // Find the child TreeView component and call its Hover method
+                            //await JSRuntime.InvokeVoidAsync("hover", "g", childNode.Neuron.Id, true, "highlightChildNode");
+                            await Hover(childNode, true,childNode.Neuron.Id);
+                            // foreach(var childNode2 in childNode.Children)
+                            //     await JSRuntime.InvokeVoidAsync("hover", "g", childNode2.Neuron.Id, true, "highlightChildNode");
+                        }
+                    }
+                }
+
+            }
+            else
+            {
+                var newChild = this.Children.FirstOrDefault(x => x.Children.Contains(child));
+                if (!Recursive)
+                {
+                    await JSRuntime.InvokeVoidAsync("hover", "g", newChild.Neuron.Id, true, "highlightNode");
+                    //await OnHover.InvokeAsync(newChild);
+                }
+                else
+                {
+                    if (child.ExpansionState == ExpansionState.Expanded && child.Children != null && child.Children.Any())
+                    {
+                        foreach (var childNode in child.Children)
+                        {
+                            // Highlight each child node 61dbc74f-ba49-4c79-9854-d4fcc3e070ad
+                            if (childNode.Neuron.Id == Id || childNode.Neuron.Type == Library.Common.RelativeType.Presynaptic)
+                                continue;
+                            await JSRuntime.InvokeVoidAsync("hover", "g", childNode.Neuron.Id, true, "highlightChildNode");
+
+                            // Recursively highlight children of children if they're expanded
+                            if (childNode.ExpansionState == ExpansionState.Expanded && childNode.Children != null && childNode.Children.Any())
+                            {
+                                // Find the child TreeView component and call its Hover method
+                                //await JSRuntime.InvokeVoidAsync("hover", "g", childNode.Neuron.Id, true, "highlightChildNode");
+                                await Hover(childNode, true, childNode.Neuron.Id);
+                                // foreach (var childNode2 in childNode.Children)
+                                //     await JSRuntime.InvokeVoidAsync("hover", "g", childNode2.Neuron.Id, true, "highlightChildNode");
+                            }
+                        }
+                    }
+                }            
+            }
+            return "";
+        }
+        catch(Exception ex)
+        {
+            var x = ex.Message;
+            return x;
+        }
+    }
+
+    private async Task<string> UnHover(TreeNeuronViewModel child, bool Recursive = false, string Id = "")
+    {
```

Comment:
UnHover should be capitalized as Unhover. Use camel casing for parameters.

Reply:
Done

Reply:
Please use camel casing for parameters.

Reply:
Done